### ISSUE TITLE
Is there a way to show the displayName of the parent ReactElement?

### ISSUE THREAD

**Issue by jergason** [rackt/react-a11y#4]
Right now the warnings are contextless - if you throw this in to an existing app with many accessibility issues, it can be hard to track down where in the codebase the problems are coming from without throwing. Is there a way to log the warnings with the `displayName` of the containing ReactElement?

**Comment by koulmomo** [comment]
just a suggestion, maybe enable the option to log a [`console.trace()`](https://developer.mozilla.org/en-US/docs/Web/API/Console.trace) along with the warning?

**Comment by jergason** [comment]
If you set `throws: true` then you can follow the stack trace as well but that is less than ideal since it slays your app.



—

**Comment by mdressman** [comment]
I took a different approach before seeing this issue and PR.  I've submitted #11 which includes the element's props in the warning/error message rather than logging a full stack trace.

**Comment by ryanflorence** [comment]
I do `throw: true` when I don't already have the offending code in my head, fix it, then put it back to not throw.

Maybe we have information about the element? I don't know, since the whole thing just hijacks `React.createElement`, we're not even in a render tree. `this.constructor.type.displayName` or something?

I don't have time right now to try to figure it out, but would happily merge a PR that does.

**Comment by jergason** [comment]
I turned on throw: true and encountered one hundred billion errors, because we did bad things. It made it hard to get to checking the components I was interested in at that time.



—

**Comment by AsaAyers** [comment]
I looked into this last night. Here is an example from React's home page.

```
var HelloMessage = React.createClass({
  render: function() {
    return <div>Hello {this.props.name}</div>;
    // return React.createElement("div", null, "Hello ", this.props.name);
  }
});

React.render(<HelloMessage name="John" />, mountNode);
// React.render(React.createElement(HelloMessage, {name: "John"}), mountNode);
```

Because they are two separate calls to `React.createElement`, there's no way to reliably associate the `div` above with `HelloMessage`. `div` is not one of the children when creating `HelloMessage`, so we can't find it from `react-a11y`.

If the code were written like this we would get access to the `displayName`, but writing all your code in one giant template defeats the purpose of components.

```
React.render(<HelloMessage name="John">
	<div>Hello {this.props.name}</div>
</HelloMessage>, mountNode);
```

My solution
=========

What I came up with is to actually set an `id` on any elements with warnings (if one doesn't already exist) and include that ID in the message. #14 

     #a11y-0 You have an anchor with `href="#"` and no `role` DOM property. Add `role="button"` or better yet, use a `<button/>`.

**Comment by kloots** [comment]
Closing as I've just merged @AsaAyers's solution in PR #14
